src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table/table.component.ts

```diff
@@ -509,20 +510,22 @@ export class TableComponent implements AfterContentChecked, OnInit, OnChanges, O
     return this.subSearch(data, currentSearch, columnsClone);
   }
 
-  basicDataSearch(searchTerm: string, data: any[], columns: CdTableColumn[]) {
+  basicDataSearch(searchTerm: string, rows: any[], columns: CdTableColumn[]) {
     if (searchTerm.length === 0) {
-      return data;
+      return rows;
     }
-    return data.filter((d) => {
+    return rows.filter((row) => {
       return (
-        columns.filter((c) => {
-          let cellValue: any = _.get(d, c.prop);
-          if (!_.isUndefined(c.pipe)) {
-            cellValue = c.pipe.transform(cellValue);
+        columns.filter((col) => {
+          let cellValue: any = _.get(row, col.prop);
+
+          if (!_.isUndefined(col.pipe)) {
+            cellValue = col.pipe.transform(cellValue);
           }
           if (_.isUndefined(cellValue)) {
```

Comment:
The UI crashes below when ``cellValue`` is ``null``. Shouldn't we exit here immediately?.

![auswahl_002](https://user-images.githubusercontent.com/1897962/53799767-729ca680-3f3b-11e9-96c8-26a65a0dec33.png)

Reply:
I'm not sure how to reproduce it, can you please explain how to do it?

And, am I correctly assuming that you are referring to the implementation already in master, as I've only refactored variable names in this code, but not the functionality?

Reply:
I've an up-to-date master and pulled your PR. After that i created a replicated pool and two RBD's as you can see in the upper screenshot.

![auswahl_003](https://user-images.githubusercontent.com/1897962/53801205-dbd1e900-3f3e-11e9-97da-bdcf358a5df2.png)

Finally i only typed a ``t`` in the search field.

Reply:
You found a bug. I created a separate [issue](http://tracker.ceph.com/issues/38583) for this as this problem has been reproducible on master. There's also a [pull request](https://github.com/ceph/ceph/pull/26766) with your suggested solution.